### PR TITLE
Update course enrollment to handle email changes

### PR DESCRIPTION
**Pull request by skaex**
### Description
This allows for course enrollment process to handle email changes on Salesforce

### Motivation and Context
Currently, when a user changes their email on Salesforce, platform doesn't handle this very well. This is causing an issue when syncing to LMS. 

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] My code follows the code style of this project.
- [x] I have added neccessary labels
- [x] My PR title follows the convention in the guide
- [x] I have assigned myself to the PR
- [x] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/canvas_api.rb [hunk 1]
```diff
@@ -118,14 +118,20 @@ def find_user_by!(email:)
...
 
-    LMSUser.new(user['id'])
+    LMSUser.new(user['id'], user['email'])
   end
 
-  def find_user_by(email:)
+  def find_user_by(email:, salesforce_contact_id:, student_id:)
     user = find_user_in_canvas(email)
+    if user.nil?
+      user = find_user_in_canvas(
```
Comment (sadleb) [inline lib/canvas_api.rb:127]: Does this call work with the SISID? If so, can we change the argument name of the method from `email` to `email_or_sis_id` ?

File: lib/salesforce_api.rb [hunk 2]
```diff
@@ -163,7 +163,7 @@ def find_contact(id:)
   def find_participants_by(program_id:)
     participants = get_participants(program_id)
 
-    particpants.map do |participant|
+    participants.map do |participant|
```
Comment (sadleb) [inline lib/salesforce_api.rb:166]: So I guess this wasn't doing anything before ?!?

**Review by sadleb — CHANGES_REQUESTED** [review]

**Comment by sadleb** [comment]
We should add specs before merging.

File: app/services/sync_portal_enrollments_for_program.rb [hunk 3]
```diff
@@ -26,6 +32,16 @@ def run
...
+  def email_inconsistent?(portal_user, participant)
+    !portal_user.email.eql?(participant.email)
+  end
+
+  def reconcile_email!(portal_user, participant)
+    ReconcileUserEmail.new(salesforce_participant: participant,
+                           portal_user: portal_user)
+                      .run
+  end
+
```
[the remark this answers is not in the document]
Comment (skaex) [reply app/services/sync_portal_enrollments_for_program.rb:44]: I want to be able to run reconciliation without doing a sync of enrollment in the future.

**Pull request merged**